python-sdk/nuscenes/lidarseg/lidarseg_utils.py

```diff
@@ -0,0 +1,228 @@
+# nuScenes dev-kit.
+# Code written by Fong Whye Kit, 2020.
+
+from typing import Dict, Iterable, List, Tuple
+
+import cv2
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+
+
+def get_stats(points_label: np.array, num_classes: int) -> List[int]:
+    """
+    Get frequency of each label in a point cloud.
+    :param num_classes: The number of classes.
+    :param points_label: A numPy array which contains the labels of the point cloud; e.g. np.array([2, 1, 34, ..., 38])
+    :return: An array which contains the counts of each label in the point cloud. The index of the point cloud
+              corresponds to the index of the class label. E.g. [0, 2345, 12, 451] means that there are no points in
+              class 0, there are 2345 points in class 1, there are 12 points in class 2 etc.
+    """
+
+    lidarseg_counts = [0] * num_classes  # Create as many bins as there are classes, and initialize all bins as 0.
+
+    indices: np.ndarray = np.bincount(points_label)
+    ii = np.nonzero(indices)[0]
+
+    for class_idx, class_count in zip(ii, indices[ii]):
+        lidarseg_counts[class_idx] += class_count  # Increment the count for the particular class name.
+
+    return lidarseg_counts
+
+
+def get_key_from_value(dictionary: dict, target_value: str):
+    """
+    Get the key belonging to a desired value in a dictionary. If there are multiple values which match the target_value,
+    only the first key will be returned.
+    :param dictionary: The dictionary to act on.
+    :param target_value: The desired value.
+    """
+    return list(dictionary.keys())[list(dictionary.values()).index(target_value)]
```

Comment:
This looks complicated. Maybe we should separate it to two lines.

Reply:
Deleted function